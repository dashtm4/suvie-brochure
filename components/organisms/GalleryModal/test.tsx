import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { GalleryModal } from '.'

describe('<GalleryModal />', () => {
  it('should renders correct <ProductCardModalContent />', () => {
    const toggleModal = jest.fn()
    const children = 'test'

    const wrapper = shallow(
      <GalleryModal toggleModal={toggleModal}>{children}</GalleryModal>
    )

    expect(wrapper.contains(children)).toBeTruthy()

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should has gallery from prop', () => {
    const toggleModal = jest.fn()
    const children = 'test'
    const gallery = ['image-1.jpg', 'imag2.jpg']

    const wrapper = shallow(
      <GalleryModal toggleModal={toggleModal} gallery={gallery}>
        {children}
      </GalleryModal>
    )

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
