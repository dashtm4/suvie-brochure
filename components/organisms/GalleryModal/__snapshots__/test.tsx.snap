// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GalleryModal /> should has gallery from prop 1`] = `
<Modal
  toggleModal={[MockFunction]}
>
  <div
    className="md:flex"
    style={
      Object {
        "minHeight": "37rem",
      }
    }
  >
    <div
      className="bg-secondary md:w-1/2 md:order-1"
      style={
        Object {
          "height": "0px",
        }
      }
    >
      <div
        className="relative w-full"
        style={
          Object {
            "height": "0px",
          }
        }
      >
        <Image
          alt=""
          layout="fill"
          objectFit="cover"
          src="image-1.jpg"
        />
      </div>
    </div>
    <div
      className="py-8 px-3 md:w-1/2 md:py-12 md:pl-8 md:pr-12 md:order-0 md:flex md:flex-col md:justify-center"
    >
      test
    </div>
  </div>
</Modal>
`;

exports[`<GalleryModal /> should renders correct <ProductCardModalContent /> 1`] = `
<Modal
  toggleModal={[MockFunction]}
>
  <div
    className="md:flex"
    style={
      Object {
        "minHeight": "37rem",
      }
    }
  >
    <div
      className="bg-secondary md:w-1/2 md:order-1"
      style={
        Object {
          "height": "0px",
        }
      }
    >
      <div
        className="relative w-full"
        style={
          Object {
            "height": "0px",
          }
        }
      >
        <Image
          alt=""
          layout="fill"
          objectFit="cover"
          src=""
        />
      </div>
    </div>
    <div
      className="py-8 px-3 md:w-1/2 md:py-12 md:pl-8 md:pr-12 md:order-0 md:flex md:flex-col md:justify-center"
    >
      test
    </div>
  </div>
</Modal>
`;
